Chapters/Common/StringAlgorithms.cs

```diff
@@ -0,0 +1,315 @@
+        //}
+
+        public static StringAlgorithms Create(Chapter9TaskType type, object param = null)
+        {
+            switch (type)
+            {
+                case Chapter9TaskType.FromString:
+                    return new StringAlgorithms(param as string);
+
+                //case Chapter9TaskType.FromFile:
+                //    return new StringAlgorithms(param as Stream);
+
+                default:
+                    throw new ArgumentOutOfRangeException(type.GetType().Name, type, null);
+            }
+
+        }
+
+        //1
+        public int PunctuationCount()
+        {
+            int sum = 0;
+            foreach (var s in Input)
+                if(char.IsPunctuation(s))
+                        sum++;
+            return sum;
+        }
+
+        //2
+        public int PointExclamationPointCount()
+        {
+            int sum = 0;
+            foreach (var c in Input)
+                if (c == '.' || c == '!')
+                    sum++;
+            return sum;
+        }
+        
+        //3
+        public void RemoveAllCommas()
+        {
+            for (var i = 0; i < Input.Length; i++)
+                if (Input[i] == ',')
+                    Input.Remove(i, 1);
+        }
+        
+        //4
+        public void RemoveSpacesBeforePunctuation()
+        {
+            for (int i = 1; i < Input.Length; i++)
+                if (Input[i] == ' ' && char.IsPunctuation(Input[i - 1]))
+                    Input.Remove(i, 1);
+        }
+        
+        //5
+        public void InsertNumberAfterSpaces(int num)
+        {
+            for (int i = 0; i < Input.Length - 1; i++)
+                if (Input[i] == ' ')
+                    Input.Insert(i + 1, num.ToString());
+        }
+        
+        //6
+        public void InsertNumberBeforeExclamation(int num)
+        {
+            for (int i = 1; i < Input.Length; i++)
+                if (Input[i] == '!')
+                    Input.Insert(i, num.ToString());
+        }
+
+        //7
+        public int SumAllNumbers()
+        {
+            int step = 1;
+            int sum = 0;
+            int start = 0;
+            bool isFirst = false;
+
+            for (int i = 0; i < Input.Length; i+=step)
+            {
+
+                if (char.IsDigit(Input[i]) && !isFirst)
+                {
+                    isFirst = true;
+                    start = i;
+                }
+                else
+                {
+                    isFirst = false;
+                    step = i - start;
+                    
+                    byte[] num = new byte[step];
+                    for (int j = start; j < i; j++)
+                        num[i - j] = Convert.ToByte(Input[j]);
+                    for (int j = num.Length - 1; j > 0; j--)
+                        sum += num[j] * (int)Math.Pow(10, num.Length - 1 - j);
+                }
+                
+            }
+            return sum;
```

Comment:
This method looks like very complex, try to simplify this, and delete this redundant flag.